Cargo.toml

```diff
@@ -512,6 +512,9 @@ proc-macro2 = "1.0"
 progenitor = "0.8.0"
 progenitor-client = "0.8.0"
 bhyve_api = { git = "https://github.com/oxidecomputer/propolis", rev = "fae5334bcad5e864794332c6fed5e6bb9ec88831" }
+# TODO(eliza): `propolis-client` should maybe re-export the needed types from
```

Comment:
Perhaps this is a note to yourself as you self-review, but I'd suggest either do that in propolis and then implement this fix, or don't and leave out this note.

Regardless, I think a better approach would be to (somehow) include the `ErrorCode` type in the `propolis-server.json` file so that we had a generated type for it. i.e. rather than adding a new dependency into the tree.

Reply:
Oh, that's a good suggestion, I'll look into figuring out how to add it to the OpenAPI document even though it doesn't actually appear in any of the OpenAPI endpoints directly. 

That said, I'd prefer to do that in a separate change as I think fixing the currently broken behavior is urgent enough that I'd like to land this PR as soon as possible, to make sure it gets in for the release. I've removed the comment for now, and then I'll try to get this into the generated client and come back and remove then dep.

Reply:
AFAICT, adding schemas for additional types to a Dropshot-generated OpenAPI document will probably require re-parsing the JSON, sticking the new schema in the correct place, and re-serializing it. I'm not really sure if it's worth the effort to do that just for this one case when we could, alternatively, spend that time actually changing Dropshot to allow specifying error body types more generally (e.g. oxidecomputer/dropshot#39, oxidecomputer/dropshot#41)?

Reply:
Opened https://github.com/oxidecomputer/propolis/issues/775 to track this.